### PR TITLE
add receipts logs

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: migration/db/migrate/20180608083123_add_receipts_logs.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      t.binary :event_name, :limit => 32, :null => false
+      t.binary :topic1, :limit => 32, :null => true
+      t.binary :topic2, :limit => 32, :null => true
+      t.binary :topic3, :limit => 32, :null => true
+      t.binary :data, :limit => 1.megabyte, :null => false
+    end
+    add_index :receipt_logs, :tx_hash
+    add_index :receipt_logs, :block_number
+    add_index :receipt_logs, [:block_number, :contract_address]
+    add_index :receipt_logs, [:block_number, :contract_address, :event_name], :name => 'index_number_address_event_name'
```
Comment (kimmylin) [inline migration/db/migrate/20180608083123_add_receipts_logs.rb]: make it the same prefix, like "index_receipt_logs_on_nae"

File: migration/db/migrate/20180608083123_add_receipts_logs.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class AddReceiptsLogs < ActiveRecord::Migration[5.2]
+  def change
+    create_table :receipt_logs do |t|
+      t.binary :tx_hash, :limit => 32, :null => false
+      t.integer :block_number, :limit => 8, :null => false, :default => 0
```
Comment (kimmylin) [inline migration/db/migrate/20180608083123_add_receipts_logs.rb]: on second thought, i think it's a bit dangerous to have default to 0. think we should remove them.

File: store/transaction_receipt/transaction_receipt.go [hunk 3]
```diff
@@ -38,21 +34,53 @@ type store struct {
...
 func NewWithDB(db *gorm.DB) Store {
 	return &store{
-		db: db.Table(TableName),
+		db: db,
 	}
 }
 
 func (r *store) Insert(data *model.Receipt) error {
-	return r.db.Create(data).Error
+	// TODO: may need db transaction protection
```
Comment (kimmylin) [inline store/transaction_receipt/transaction_receipt.go]: better to wrap these in a tx.

File: store/transaction_receipt/transaction_receipt.go [hunk 4]
```diff
@@ -38,21 +34,53 @@ type store struct {
...
+			return err
+		}
+	}
+	return nil
 }
 
-func (r *store) Delete(from, to int64) (err error) {
-	err = r.db.Delete(model.Receipt{}, "block_number >= ? AND block_number <= ?", from, to).Error
-	return
+func (r *store) Delete(from, to int64) error {
```
Comment (kimmylin) [inline store/transaction_receipt/transaction_receipt.go]: just to confirm. this op is wrapped inside a tx at higher level, right?

File: migration/db/migrate/20180608083123_add_receipts_logs.rb (hunk 1 above)
Comment (markya0616) [reply migration/db/migrate/20180608083123_add_receipts_logs.rb]: OK

File: migration/db/migrate/20180608083123_add_receipts_logs.rb (hunk 2 above)
Comment (markya0616) [reply migration/db/migrate/20180608083123_add_receipts_logs.rb]: OK, also remove the default value in `transaction_receipts` and `transaction` tables.

File: store/transaction_receipt/transaction_receipt.go (hunk 3 above)
Comment (markya0616) [reply store/transaction_receipt/transaction_receipt.go]: OK

**Pull request merged**